components/Events.vue

```diff
@@ -0,0 +1,240 @@
+    </table>
+  </div>
+</template>
+
+<script>
+/* eslint-disable no-console */
+
+import fireDb from '../plugins/firebase'
+
+export default {
+  name: 'Events',
+  props: {
+    originalEvents: {
+      type: Object,
+      default() {
+        return {}
+      }
+    },
+    selectedWebsite: {
+      type: String,
+      default: ''
+    }
+  },
+  data() {
+    return {
+      addEditModal: false,
+      viewModal: false,
+      title: '',
+      order: '',
+      text: '',
+      eventLink: '',
+      learnMoreLink: '',
+      signupLink: '',
+      imageLink: '',
+      index: undefined,
+      events: this.originalEvents,
+      editing: false
+    }
+  },
+  methods: {
+    add_event() {
+      this.events[this.selectedWebsite].push({})
+      this.index = this.events[this.selectedWebsite].length - 1
+      this.addEditModal = true
+      return ''
+    },
+    view_event(index) {
+      this.title = this.events[this.selectedWebsite][index].title
+      this.order = this.events[this.selectedWebsite][index].order
+      this.text = this.events[this.selectedWebsite][index].text
+      this.eventLink = this.events[this.selectedWebsite][index].eventLink
+      // eslint-disable-next-line prettier/prettier
+      this.learnMoreLink = this.events[this.selectedWebsite][index].learnMoreLink
+      this.signupLink = this.events[this.selectedWebsite][index].signupLink
+      this.imageLink = this.events[this.selectedWebsite][index].imageLink
+      this.viewModal = true
+    },
+    edit_event(index) {
+      this.title = this.events[this.selectedWebsite][index].title
+      this.order = this.events[this.selectedWebsite][index].order
+      this.text = this.events[this.selectedWebsite][index].text
+      this.eventLink = this.events[this.selectedWebsite][index].eventLink
+      // eslint-disable-next-line prettier/prettier
+      this.learnMoreLink = this.events[this.selectedWebsite][index].learnMoreLink
+      this.signupLink = this.events[this.selectedWebsite][index].signupLink
+      this.imageLink = this.events[this.selectedWebsite][index].imageLink
+
+      this.editing = true
+      this.index = index
+      this.addEditModal = true
+    },
+    async delete_event(index) {
+      if (confirm('Are you sure you with to delete this event?')) {
+        await fireDb.delete(
+          this.selectedWebsite,
+          'Events',
+          this.events[this.selectedWebsite][index].id
+        )
+        window.location.reload(true)
+      }
+    },
+    cancel() {
+      this.title = ''
+      this.order = ''
+      this.text = ''
+      this.eventLink = ''
+      this.learnMoreLink = ''
+      this.signupLink = ''
+      this.imageLink = ''
+
+      if (this.index && !this.editing)
+        this.events[this.selectedWebsite].splice(this.index, 1)
+
+      this.index = undefined
+      this.addEditModal = false
+      this.editing = false
+    },
+    async save() {
+      this.events[this.selectedWebsite][this.index].title = this.title
```

Comment:
kinda worried that it might be an error when index is undefined (which is the default value)
Probably give it a check?

Reply:
this.index will always be defined. The only time it is used is in save, which can only be accessed from the modal. this.index will be defined when the modal is opened

Reply:
OK, then it looks ok